src/network/networkd-route.c

```diff
@@ -436,6 +437,10 @@ int route_configure(Route *route, Link *link,
         assert(link->manager);
         assert(link->manager->rtnl);
         assert(link->ifindex > 0);
+
+        if (route->family == AF_UNSPEC)
+                route->family = AF_INET;
```

Comment:
Hmm... Why are we falling back to IPv4? Why not IPv6? Why do we need to fall back at all? What about just allowing Gateway=0.0.0.0 or Gateway=:: ?